dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -297,22 +369,75 @@ internal constructor(
         return headerContext ?: tagContext
     }
 
+    private fun setSampledOutHeaders(
+        requestBuilder: okhttp3.Request.Builder,
+        tracingHeaderTypes: Set<TracingHeaderType>,
+        span: Span
+    ) {
+        for (headerType in tracingHeaderTypes) {
+            when (headerType) {
+                TracingHeaderType.DATADOG -> {
+                    listOf(
+                        DATADOG_SAMPLING_PRIORITY_HEADER,
+                        DATADOG_TRACE_ID_HEADER,
+                        DATADOG_SPAN_ID_HEADER,
+                        DATADOG_ORIGIN_HEADER
+                    ).forEach {
+                        requestBuilder.removeHeader(it)
+                    }
+                    requestBuilder.addHeader(
+                        DATADOG_SAMPLING_PRIORITY_HEADER,
+                        DATADOG_DROP_SAMPLING_DECISION
+                    )
+                }
+
+                TracingHeaderType.B3 -> {
+                    requestBuilder.removeHeader(B3_HEADER_KEY)
+                    requestBuilder.addHeader(B3_HEADER_KEY, B3_DROP_SAMPLING_DECISION)
+                }
+
+                TracingHeaderType.B3MULTI -> {
+                    listOf(
+                        B3M_TRACE_ID_KEY,
+                        B3M_SPAN_ID_KEY,
+                        B3M_SAMPLING_PRIORITY_KEY
+                    ).forEach {
+                        requestBuilder.removeHeader(it)
+                    }
+                    requestBuilder.addHeader(B3M_SAMPLING_PRIORITY_KEY, B3M_DROP_SAMPLING_DECISION)
+                }
+
+                TracingHeaderType.TRACECONTEXT -> {
+                    requestBuilder.removeHeader(W3C_TRACEPARENT_KEY)
+                    requestBuilder.addHeader(
+                        W3C_TRACEPARENT_KEY,
+                        W3C_DROP_SAMPLING_DECISION.format(
+                            span.context().toTraceId(),
+                            span.context().toSpanId()
+                        )
```

Comment:
ah, basically we need to have `isSampled` parameter in addition to non-null Span (even if it was sampled out) just for this case?